modules/honzon-distribution/src/lib.rs

```diff
@@ -0,0 +1,329 @@
+		let account_id = stable_asset.account_id;
+
+		let total_supply = pool_info.total_supply;
+		let ausd_supply = pool_info.balances[stable_asset.stable_token_index as usize];
+		let asset_length = pool_info.assets.len();
+
+		let current_rate = Ratio::saturating_from_rational(ausd_supply, total_supply);
+		let target_rate = params
+			.target_min
+			.saturating_add(params.target_max)
+			.saturating_mul(Ratio::saturating_from_rational(1, 2));
+		let one: Ratio = One::one();
+		let remain_rate = one.saturating_sub(target_rate);
+		let remain_reci = one.div(remain_rate);
+		if current_rate < params.target_min {
+			let numerator = target_rate.saturating_mul_int(total_supply).saturating_sub(ausd_supply);
+			let mint_amount = remain_reci.saturating_mul_int(numerator).min(params.max_step);
+
+			// should failed if after this mint, the balance of distributed exceed the capacity.
+			if let Some(exist_minted) = DistributedBalance::<T>::get(&destination) {
+				let newly_amount = mint_amount.saturating_add(exist_minted);
+				ensure!(newly_amount < params.capacity, Error::<T>::ExceedCapacity);
+			} else {
+				ensure!(mint_amount < params.capacity, Error::<T>::ExceedCapacity);
+			}
+
+			log::info!(target: "honzon-dist", "current:{:?}, target:{:?}, mint:{:?}", current_rate, target_rate, mint_amount);
+			let mut assets = vec![0; asset_length];
+			assets[stable_asset.stable_token_index as usize] = mint_amount;
+			// deposit stable asset to treasury account.
+			T::Currency::deposit(stable_asset.stable_currency_id, &account_id, mint_amount)?;
+			// use this treasury amount and mint to stable asset pool
+			T::StableAsset::mint(&account_id, stable_asset.pool_id, assets, 0)?;
+
+			Pallet::<T>::deposit_event(Event::<T>::AdjustDestination {
+				destination,
+				stable_currency: stable_asset.stable_currency_id,
+				amount: mint_amount as Amount,
+			});
+			return Ok(mint_amount as Amount);
+		} else if current_rate > params.target_max {
+			let numerator = ausd_supply.saturating_sub(target_rate.saturating_mul_int(total_supply));
+			let burn_amount = remain_reci.saturating_mul_int(numerator).min(params.max_step);
+			log::info!(target: "honzon-dist", "current:{:?}, target:{:?}, burn:{:?}", current_rate, target_rate, burn_amount);
+			let (_, stable_amount) = T::StableAsset::redeem_single(
+				&account_id,
+				stable_asset.pool_id,
+				burn_amount,
+				stable_asset.stable_token_index,
+				0,
+				asset_length as u32,
+			)?;
+			T::Currency::withdraw(stable_asset.stable_currency_id, &account_id, stable_amount)?;
+			let burn_amount = (0 as Amount).saturating_sub(stable_amount as Amount);
+
+			Pallet::<T>::deposit_event(Event::<T>::AdjustDestination {
+				destination,
+				stable_currency: stable_asset.stable_currency_id,
+				amount: burn_amount,
+			});
+			return Ok(burn_amount);
+		}
+
+		Ok(0 as Amount)
+	}
+}
+
+pub struct RebasedStableAssetErrorConvertor<T>(PhantomData<T>);
```

Comment:
this is only used for test? should move to mock

Reply:
move to module_support/mock.